File: 06-base-ps-java/src/main/java/com/algopatterns/easy/IntersectionOfTwoArrays.java
package com.algopatterns.easy;

import java.util.Arrays;

public class IntersectionOfTwoArrays {
    public static int[] intersection(int[] nums1, int[] nums2) {
        // 여기에 코드를 작성하세요.
        
        return new int[0];
    }

    public static void main(String[] args) {
        // 첫 번째 테스트 케이스를 위한 배열들을 생성합니다.
        int[] case1Nums1 = {1, 2, 2, 1};
        int[] case1Nums2 = {2, 2};
        // 교집합 결과를 배열 형태로 반환받아 출력합니다.
        System.out.println(Arrays.toString(intersection(case1Nums1, case1Nums2))); // 예상 결과: [2]

        // 두 번째 테스트 케이스를 위한 배열들을 생성합니다.
        int[] case2Nums1 = {4, 9, 5};
        int[] case2Nums2 = {9, 4, 9, 8, 4};
        // 교집합 결과를 배열 형태로 반환받아 출력합니다.
        System.out.println(Arrays.toString(intersection(case2Nums1, case2Nums2))); // 예상 결과: [4, 9] (순서 무관)
    }
}

File: daily-practice/intersection_of_two_arrays.py
def intersection(nums1, nums2):
    # 여기에 코드를 작성하세요.
    # 첫번째 배열을 중복이 없는 셋으로 변환
    set1 = set(nums1)
    set2 = set(nums2)

    # 두 셋의 교집합(&) 을 구한뒤, 이를 다시 리스트(list)형으로 변환하여 반환
    return list(set1 & set2)

# 테스트 케이스
if __name__ == "__main__":
    case1_nums1 = [1, 2, 2, 1]
    case1_nums2 = [2, 2]
    print(intersection(case1_nums1, case1_nums2))  # 예상 결과: [2]

    case2_nums1 = [4, 9, 5]
    case2_nums2 = [9, 4, 9, 8, 4]
    print(intersection(case2_nums1, case2_nums2))  # 예상 결과: [9, 4] (또는 [4, 9])
